### ISSUE TITLE
Rnn generate command returns the input text

### ISSUE THREAD

**Issue by loretoparisi** [pjreddie/darknet#210]
I have trained a new textual rnn with the command

`./darknet rnn train cfg/rnn.train.cfg -file /root/my_text_dataset.txt`

I have used the default `rnn.train.cfg` configuration, with `batch=64` and `steps=1000,1500`.

The generation command was like

`./darknet rnn generate cfg/rnn.cfg ./my_model.weights -srand 0 -len 1000 -seed love`

This returns part of the text in the dataset (the exact text, i.e. not a new generated text) starting from the seed.

**Comment by loretoparisi** [comment]
*[UPDATE]*

I have tried to train a new model weights for `shakespeare` text. Following the generated text from my trained model

```
# ./darknet rnn generate cfg/rnn.cfg /root/rnn_shakespeare_model.weights -srand 0 -len 1000 -seed BRUTUS
rnn
layer     filters    size              input                output
    0 RNN Layer: 256 inputs, 1024 outputs
		connected                             256  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    1 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    2 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    3 connected                            1024  ->   256
    4 softmax                                         256
    5 cost                                            256
Loading weights from /root/rnn_shakespeare_model.weights...Done!
BRUTUS. Mark him out of a curse?
    The second of it be but young in a child, and of his benediction
    shall never have been a man transform your hands.
    It add so admired them.
  JULIA. Why should you be
    The bloody mind be gone to use?
    And then I would not throw upon your honour.
    To-morrow to my love with self-four kingdom,
    Whose eye-besides the means of his across sounds,
    And made them hot to die to her his face,
    The rest and the sword which else can be much.  
    I know you hear, my lord.
  DUKE. Where is he? Why, this is not a horse,
    Or else he died.
  ROSALIND. Dear lords, with their love shall die presently.
    There is no sea of fear.
  GLOUCESTER. The sound that's coming for a drum.
    Thou wast so dark'ned in your own prophecy,
    The gates of Helen's sake, and hast thou lov'd.
    The house is not so bad as well awhile.
    My virtue is ours of our worthy sweet;
    Seek him and never break out them in you.
    We are our chains, and cure the m
```

and here the generated text from the web site weights:

```
# ./darknet rnn generate cfg/rnn.cfg ./shakespeare.weights -srand 0 -len 1000 -seed BRUTUS
rnn
layer     filters    size              input                output
    0 RNN Layer: 256 inputs, 1024 outputs
		connected                             256  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    1 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    2 RNN Layer: 1024 inputs, 1024 outputs
		connected                            1024  ->  1024
		connected                            1024  ->  1024
		connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    3 connected                            1024  ->   256
    4 softmax                                         256
    5 cost                                            256
Loading weights from ./shakespeare.weights...Done!
BRUTUS. Now, but he poison with the world.
    So I near me, you know he is now tender off
    The fearful gates, and heart in hand a sweet  
     Perchance shuts a d the Prince of Wales,
    Witness which he bade esteem'd it.
    And then she would be as mine own ends,
    Whose designs which the mouth's outward sons.
    I am sorry for your blood, the shadow of my part,
    Whose grief shall be the battle to a fine
    Of the most infection of a spite
    Whereon thy self and fall the struant;
    And when they strike this morning to our love.
    Therefore I love thee let me with a power
    Of such a time with your displeasure that
    And he shall stay for no ready.
    My brother wears your queen in love by sea,
    Which is my friends and hearts now dear in their
                                                                  Exeunt




SCENE II.
The frontier was this forest

Enter CLEOPATRA, LOVEL, and LORDS

  CAMILLO. A cursed love to my will.
  Beat. I thank it out that I can se
```

**Comment by loretoparisi** [comment]
@pjreddie any idea why the inference reproduces the inputs?

**Comment by CBIR-LL** [comment]
Hi, can I ask you a question about input text? Unable to download and reference the Shakespeare.txt, so I don't know how to make my own train.txt. Can you show me your input file ? Thank you very much.

**Comment by loretoparisi** [comment]
@CBIR-LL see here for an example how to use the `rnn` for generation: https://github.com/loretoparisi/docker/tree/master/darknet 
The generator works properly from the pre-trained weights, while the training of a new model is described above, but currently I cannot set the right parameter to generate new samples correctly from a brand new model.

**Comment by CBIR-LL** [comment]
Thanks a lot. 
`
[lilai@host01 darknet-17.11.28]$ ./darknet rnn generate cfg/rnn.cfg backup_rnn/rnn_final.weights -seed BRUTUS
rnn
layer     filters    size              input                output
    0 RNN Layer: 256 inputs, 1024 outputs
                connected                             256  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    1 RNN Layer: 1024 inputs, 1024 outputs
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    2 RNN Layer: 1024 inputs, 1024 outputs
                connected                            1024  ->  1024
                connected                            1024  ->  1024
                connected                            1024  ->  1024
Unused field: 'hidden = 1024'
    3 connected                            1024  ->   256
    4 softmax                                         256
    5 cost                                            256
Loading weights from backup_rnn/rnn_final.weights...Done!
BRUTUS:
To hear and work at home.

GLOUCESTER:

PROTEUS:
And I, to make a valiant cause:
If thou art ruled by the uncle Thoras
That will unto the bearing of the world:
She is his father's death of love and made
A little constant daughter. Heaven keep your mild!

HORATIO:
It is in love, they will be denied your lordship.

PROSPERO:
So they too.

NORFOLK:
That you do not sleep, that I am bound to fear,
And she is made a man to blame your ways.

SALARINO:
I have no remedy.

KING HENRY VI:
My lord, I beseech you, sir; I will not drunk you
To the better than a fair lady's crown.

PERICLES:
The laws of John a beard of Britain so
That we should every over-maid she shall,
And she have bought them; and these fortunes have
The what of her young man was false to hell;
And now that I was time to add my shoulders;
The greatest proof with clamorous enemies,
What beauty forged me with thy hands of soul
And find him that have said and spend his end,
As my bosoms to the gates of Nine
May be a very dear frien
[lilai@host01 darknet-17.11.28]$
`

**Comment by CBIR-LL** [comment]
Thanks a lot. I have trained it and follwing is my output.
```
 ./darknet rnn generate cfg/rnn.cfg backup_rnn/rnn_final.weights -seed BRUTUS

BRUTUS:
To hear and work at home.

GLOUCESTER:

PROTEUS:
And I, to make a valiant cause:
If thou art ruled by the uncle Thoras
That will unto the bearing of the world:
She is his father's death of love and made
A little constant daughter. Heaven keep your mild!

HORATIO:
It is in love, they will be denied your lordship.

PROSPERO:
So they too.

NORFOLK:
That you do not sleep, that I am bound to fear,
And she is made a man to blame your ways.

SALARINO:
I have no remedy.

KING HENRY VI:
My lord, I beseech you, sir; I will not drunk you
To the better than a fair lady's crown.

PERICLES:
The laws of John a beard of Britain so
That we should every over-maid she shall,
And she have bought them; and these fortunes have
The what of her young man was false to hell;
And now that I was time to add my shoulders;
The greatest proof with clamorous enemies,
What beauty forged me with thy hands of soul
And find him that have said and spend his end,
As my bosoms to the gates of Nine
May be a very dear frien

```

**Comment by loretoparisi** [comment]
@CBIR-LL great it seems that it's generating new samples! Which was your **rnn.train.cfg** and train command? 
I did `./darknet rnn train cfg/rnn.train.cfg -file ./t8.shakespeare.txt `, but I'm not sure if I have modified the configuration it in the right way (see above) about the lstm part...

**Comment by CBIR-LL** [comment]
@loretoparisi I used the rnn.train.cfg and rnn.cfg provided by the author. 
'''
rnn.train.cfg
[net]
subdivisions=1
inputs=256###256
batch = 64###128
momentum=0.9
decay=0.001
max_batches = 2000
time_steps=576
learning_rate=0.1
policy=steps
steps=1000,1500
scales=.1,.1

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[connected]
output=256
activation=leaky

[softmax]

[cost]
type=sse


'''

'''
[net]
subdivisions=1
inputs=256
batch = 1
momentum=0.9
decay=0.001
max_batches = 2000
time_steps=1
learning_rate=0.1
policy=steps
steps=1000,1500
scales=.1,.1

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[rnn]
batch_normalize=1
output = 1024
hidden=1024
activation=leaky

[connected]
output=256
activation=leaky

[softmax]

[cost]
type=sse


'''

**Comment by loretoparisi** [comment]
@CBIR-LL ok so you changed to `batch` size to 64 from the default configuration (where it was like 128). In fact this will fix the `darknet: ./src/cuda.c:36: check_error: Assertion `0' failed.` error.
I'm trying to re-run the training again this time using the AlexeyAB fork - see https://github.com/AlexeyAB/darknet/issues/286#issuecomment-351349341
